domain/charm/doc.go

```diff
@@ -0,0 +1,8 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package charm provides the domain types for a charm. The application domain
+// is the primary domain for creation and handling of a charm. The types here
+// are to ensure that the charm is handled correctly and that the charm is
+// correctly represented in the domain.
+package charm
```

Comment:
This should be deleted/merged with the application domain one. Also the types.go should be migrated over. Theres a lot of functions in application that still have its types, hopefully some IDE magic can help.

If the types don't belong in application then maybe they could be moved to core or somewhere like that, since its a bit weird to have a domain just for some types.

 Looks like once you've done that you can just delete the charm directory, no reason to keep it around.